velox/exec/HashTable.h

```diff
@@ -949,6 +958,9 @@ class HashTable : public BaseHashTable {
   // Owns the memory of multiple build side hash join tables that are
   // combined into a single probe hash table.
   std::vector<std::unique_ptr<HashTable<ignoreNullKeys>>> otherTables_;
+
+  std::unordered_map<char*, std::shared_ptr<std::vector<char*>>> duplicateRows_;
```

Comment:
It's so weird that we use another hash table inside a hash table.  Can we create some space in row container to keep this information for each slot instead of probing another hash table?  We will need to be careful when we erase rows in this case though.